resources/assets/lib/chat/chat-orchestrator.ts

```diff
@@ -157,28 +158,38 @@ export default class ChatOrchestrator implements DispatchListener {
   }
 
   markAsRead(channelId: number) {
-    const channel = this.rootDataStore.channelStore.getOrCreate(channelId);
-    const lastReadId = channel.lastMessageId;
-
-    if (!channel.isUnread) {
+    if (this.markingAsRead[channelId] != null) {
       return;
     }
 
     // We don't need to send mark-as-read for our own messages, as the cursor is automatically bumped forward server-side when sending messages.
-    const lastSentMessage = channel.messages[channel.messages.length - 1];
-    if (lastSentMessage && lastSentMessage.sender.id === window.currentUser.id) {
-      channel.lastReadId = lastReadId;
+    const channel = this.rootDataStore.channelStore.getOrCreate(channelId);
 
+    if (!channel.isUnread) {
       return;
     }
 
-    this.api.markAsRead(channel.channelId, lastReadId)
-      .then(() => {
+    this.markingAsRead[channelId] = window.setTimeout(() => {
+      const lastReadId = channel.lastMessageId;
+
+      const lastSentMessage = channel.messages[channel.messages.length - 1];
+      if (lastSentMessage && lastSentMessage.sender.id === window.currentUser.id) {
         channel.lastReadId = lastReadId;
-      })
-      .catch((err) => {
-        console.debug('markAsRead error', err);
-      });
+
+        return;
```

Comment:
need to unset the channel from `markingAsRead` here as well or it'll get stuck when the latest message is yours when it runs.